qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -328,18 +332,45 @@ def _calculate_clim(self) -> tuple:
     def _manual_clim(self):
         """Update stored colorbar limits
         The new limits are found from user input"""
-        if self.cmin.hasAcceptableInput():
+        self._update_cmin()
+        self._update_cmax()
+
+    def _update_cmin(self):
```

Comment:
This and `_update_cmax()` look to be very similar. Is there a way they could be re-factored to share the repeated implementation?

Reply:
I had a look at this and think its quite awkward to combine them due to the direction of the limit check changing. I think the most strictly oop way of doing things would be to extend the `qlineedit` class to make our own min/max box class which would have the boxes aware of each other. Then you could just call `update()` on each box which would make that bit of code a lot simpler but adds a lot of code elsewhere.